akka-actor/src/main/scala/akka/pattern/BackoffOnRestartSupervisor.scala

```diff
@@ -38,7 +38,15 @@ private class BackoffOnRestartSupervisor(
         // to our own Restarts, which involves stopping the child.
         case Restart ⇒
           val childRef = sender()
-          become(waitChildTerminatedBeforeBackoff(childRef) orElse handleBackoff)
+          val nextRestartCount = restartCount + 1
+          if (strategy.maxNrOfRetries >= 0 && nextRestartCount > strategy.maxNrOfRetries) {
```

Comment:
That would be great if you added. I don't think this can be useful without taking reset time into account.